src/network/networkd-link.c

```diff
@@ -515,6 +530,9 @@ static void link_free(Link *link) {
 
         free(link->ifname);
 
+        if (link->kind)
+                free(link->kind);
```

Comment:
nitpick: free() is a NOP when it gets NULL, and we make heavy use of that, to shorten our destructor code. Hence, you may drop the explicit if check here, and just call free() directly.